tests/mock-constants.php

```diff
@@ -79,3 +79,15 @@ function constant( $constant ) {
 		return Constant_Mocker::constant( $constant );
 	}
 }
+
+namespace Automattic\VIP\Cache {
```

Comment:
Is the reason we need this that the prod code is doing `defined` and `constant` directly? So this will kinda overwrite the php core functions with the namespaced versions?

Reply:
Yes :-) We inject our helpers into the respective namespace (this is why we have to duplicate the same function for every namespace), and then the production code, due to the namespace resolution rules, calls our implementation instead of the PHP's one.